server/src/main/java/org/apache/druid/server/coordinator/CuratorLoadQueuePeonV2.java

```diff
@@ -0,0 +1,418 @@
+    this.checkNodeRemovedExecutor = Executors.newScheduledThreadPool(5, new ThreadFactoryBuilder().setNameFormat(
+        "ZKNodeDeletionChecker--%d").build());
+  }
+
+  @JsonProperty
+  @Override
+  public Set<DataSegment> getSegmentsToLoad()
+  {
+    return segmentsToLoad.keySet();
+  }
+
+  @JsonProperty
+  @Override
+  public Set<DataSegment> getSegmentsToDrop()
+  {
+    return segmentsToDrop.keySet();
+  }
+
+  @JsonProperty
+  @Override
+  public Set<DataSegment> getSegmentsMarkedToDrop()
+  {
+    return segmentsMarkedToDrop;
+  }
+
+  @Override
+  public long getLoadQueueSize()
+  {
+    return queuedSize.get();
+  }
+
+  @Override
+  public int getAndResetFailedAssignCount()
+  {
+    return failedAssignCount.getAndSet(0);
+  }
+
+  @Override
+  public int getNumberOfSegmentsInQueue()
+  {
+    return segmentsToLoad.size();
+  }
+
+  @Override
+  public void loadSegment(
+      final DataSegment segment,
+      final LoadPeonCallback callback
+  )
+  {
+
+    final SegmentHolder existingHolder = segmentsToLoad.get(segment);
+    if (existingHolder != null) {
+      if ((callback != null)) {
+        existingHolder.addCallback(callback);
+      }
+      return;
+    }
+    log.info("Asking server peon[%s] to load segment[%s]", basePath, segment.getId());
+    queuedSize.addAndGet(segment.getSize());
+    SegmentHolder holder = new SegmentHolder(segment, LOAD, Collections.singletonList(callback));
+    segmentsToLoad.put(segment, holder);
+    processingExecutor.submit((Callable<Void>) () -> {
+      processSegmentChangeRequest(holder);
+      return null;
+    });
+  }
+
+  @Override
+  public void dropSegment(
+      final DataSegment segment,
+      final LoadPeonCallback callback
+  )
+  {
+    final SegmentHolder existingHolder = segmentsToDrop.get(segment);
+    if (existingHolder != null) {
+      if (callback != null) {
+        existingHolder.addCallback(callback);
+      }
+      return;
+    }
+    log.info("Asking server peon[%s] to drop segment[%s]", basePath, segment.getId());
+    SegmentHolder segmentHolder = new SegmentHolder(segment, DROP, Collections.singletonList(callback));
+    segmentsToDrop.put(segment, segmentHolder);
+    processingExecutor.submit((Callable<Void>) () -> {
+      processSegmentChangeRequest(segmentHolder);
+      return null;
+    });
+  }
+
+  @Override
+  public void markSegmentToDrop(DataSegment dataSegment)
+  {
+    segmentsMarkedToDrop.add(dataSegment);
+  }
+
+  @Override
+  public void unmarkSegmentToDrop(DataSegment dataSegment)
+  {
+    segmentsMarkedToDrop.remove(dataSegment);
+  }
+
+  private void processSegmentChangeRequest(SegmentHolder segmentHolder)
+  {
+    try {
+      // Wait for a random duration since we do not want zookeeper getting pounded with create node requests
+      TimeUnit.MILLISECONDS.sleep(ThreadLocalRandom.current().nextInt(MAX_TASK_WAIT));
+      log.info("Server[%s] %s [%s]", basePath, segmentHolder.getType() == LOAD ?
```

Comment:
This is not aligned with the Druid code style (all args fit the line _or_ each arg is broken down on its own line).

Please don't use multiline ternary operator.